service/control.go

```diff
@@ -69,10 +75,36 @@ func (ctrl *Control) postWorkflow(c *gin.Context) {
 		Owner:       body.Owner,
 		IsActive:    body.IsActive,
 	}
-	ctrl.db.Create(&wf)
+	// upsert workflow
+	ctrl.db.Clauses(
+		clause.OnConflict{
+			Columns:   []clause.Column{{Name: "name"}},
+			DoUpdates: clause.AssignmentColumns([]string{"updated_at", "artifact", "command", "every", "next_runtime", "backfill", "owner", "is_active"}),
+		}).Create(&wf)
+	ctrl.db.Unscoped().Model(&wf).Update("deleted_at", nil)
 	c.JSON(http.StatusOK, "OK")
 }
 
+func (ctrl *Control) deleteWorkflow(c *gin.Context) {
+	var body deleteWorkflowReq
+	if err := c.BindJSON(&body); err != nil {
+		// bad request
+		c.JSON(http.StatusBadRequest, gin.H{"message": "could not parse body"})
+		return
+	}
+
+	dbRes := ctrl.db.Where("name = ?", body.Name).Delete(&table.Workflow{})
+	if (dbRes.Error != nil && errors.Is(dbRes.Error, gorm.ErrRecordNotFound)) || dbRes.RowsAffected == 0 {
```

Comment:
`dbRes.Error != nil && errors.Is(dbRes.Error, gorm.ErrRecordNotFound))` sounds like internal error to me

Reply:
new commit pushed